lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -141,7 +141,10 @@ private function attributeShouldBeSaved( $name, $value ) {
 	}
 
 	private function attributeShouldBeDeleted( $name, $value ) {
-		return isset( $this->defaultAttributes[$name] ) && $value == $this->defaultAttributes[$name];
+		return (
+			( isset( $this->defaultAttributes[$name] ) && $value == $this->defaultAttributes[$name] ) ||
+			is_null( $value )
```

Comment:
I don't think that logic should conflict. When it comes to default attributes, we only store them in the service if the user has a non-default value for them. Otherwise, we just pull them from the `$defaultAttributes` array which is loaded from a config. See [here](https://github.com/Wikia/app/blob/SOC-1820/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L58-L60). By setting an attribute with a default to null, we're effectively saying "delete whatever value the user had for this attribute in the service and use the default from here on out."

For example, if we set `fancySig` to null via  `UserAttribute::setAttribute`, then what gets stored in the attribute array for that user is the default value for `fancySig`, [which is 0](https://github.com/Wikia/app/blob/dev/includes/DefaultSettings.php#L3218). Then for subsequent calls to `$userAttributes->getAttribute( 'fancySig' )`, 0 would be returned.

When we came to actually saving the attributes, we'd call `attributeShouldBeDeleted()` for `fancySig` and see that the user has the default value set. So that means we should delete it from the service. That way we can update the default to something else and not have to worry about updating that value for all users. It's just a config change.

I may not be 100% understanding your question though. Does that explanation make sense?